test/string_extension_test.dart

```diff
@@ -330,4 +330,37 @@ void main() {
       expect(result, equals(text));
     });
   });
+
+  group('msisdnSanitizer tests', () {
+    test('msisdnSanitizer returns the correct msisdn with leading plus', () {
+      final result = '+08030000000'.msisdnSanitizer();
+      expect(result, equals('08030000000'));
+    });
+
+    test('msisdnSanitizer returns the correct msisdn without leading plus', () {
+      final result = '08030000000'.msisdnSanitizer();
+      expect(result, equals('08030000000'));
+    });
+
+    test('msisdnSanitizer returns the correct msisdn with special characters',
+        () {
+      final result = '+234803000#!*()%,^&0000'.msisdnSanitizer();
```

Comment:
idem

Reply:
I mean, not a good idea, when we work on sth wrong. Please ignore this case, make the user do what he wants.